components/TextControls/src/Theming/MDCFilledTextField+MaterialTheming.m

```diff
@@ -0,0 +1,195 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCFilledTextField+MaterialTheming.h"
+
+#import <Foundation/Foundation.h>
+
+#import "../../src/private/MDCBaseTextField+MDCTextControl.h"
+#import "../../src/private/MDCTextControl.h"
+#import "../../src/private/MDCTextControlStyleFilled.h"
+
+@implementation MDCFilledTextField (MaterialTheming)
+
+- (void)applyThemeWithScheme:(nonnull id<MDCContainerScheming>)containerScheme {
+  [self applyTypographyScheme:[self typographySchemeWithContainerScheme:containerScheme]];
+  [self applyDefaultColorScheme:[self colorSchemeWithContainerScheme:containerScheme]];
+}
+
+- (void)applyErrorThemeWithScheme:(nonnull id<MDCContainerScheming>)containerScheme {
+  [self applyTypographyScheme:[self typographySchemeWithContainerScheme:containerScheme]];
+  [self applyErrorColorScheme:[self colorSchemeWithContainerScheme:containerScheme]];
+}
+
+- (id<MDCColorScheming>)colorSchemeWithContainerScheme:
+    (nonnull id<MDCContainerScheming>)containerScheme {
+  id<MDCColorScheming> mdcColorScheme = containerScheme.colorScheme;
+  if (!mdcColorScheme) {
+    mdcColorScheme =
+        [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  }
+  return mdcColorScheme;
+}
+
+- (id<MDCTypographyScheming>)typographySchemeWithContainerScheme:
+    (nonnull id<MDCContainerScheming>)containerScheme {
+  id<MDCTypographyScheming> mdcTypographyScheme = containerScheme.typographyScheme;
+  if (!mdcTypographyScheme) {
+    mdcTypographyScheme =
+        [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201902];
+  }
+  return mdcTypographyScheme;
+}
+
+- (void)applyTypographyScheme:(id<MDCTypographyScheming>)mdcTypographyScheming {
+  self.font = mdcTypographyScheming.subtitle1;
+  self.leadingAssistiveLabel.font = mdcTypographyScheming.caption;
+  self.trailingAssistiveLabel.font = mdcTypographyScheming.caption;
+}
+
+- (void)applyDefaultColorScheme:(id<MDCColorScheming>)colorScheme {
+  CGFloat disabledOpacity = (CGFloat)0.60;
+
+  UIColor *textColorNormal = [colorScheme.onSurfaceColor colorWithAlphaComponent:(CGFloat)0.87];
+  UIColor *textColorEditing = textColorNormal;
+  UIColor *textColorDisabled = [textColorNormal colorWithAlphaComponent:disabledOpacity];
+
+  UIColor *assistiveLabelColorNormal =
+      [colorScheme.onSurfaceColor colorWithAlphaComponent:(CGFloat)0.60];
```

Comment:
Nit: Define this alpha values in a const?

Reply:
Done.

Reply:
Can we define all these values as constants?